File: 1299. Replace Elements with Greatest Element on Right Side/Solution.cs
public class Solution {
    public int[] ReplaceElements(int[] arr) {
        
        int greatestValue = -1;
        
        for (int i = arr.Length-1; i >= 0; --i) {
            
            int newValue = greatestValue;
            greatestValue = Math.Max(greatestValue, arr[i]);
            arr[i] = newValue;
        }
        return arr;
    }
}

File: 1299. Replace Elements with Greatest Element on Right Side/solution.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:

        greatestValue = -1
        
        for index in range(len(arr)-1,-1,-1):
            newValue = greatestValue
            greatestValue = max(greatestValue, arr[index])
            arr[index] = newValue
        
        return arr
